development/webpack/webpack.config.ts

```diff
@@ -0,0 +1,397 @@
+    data: {
+      isMMI: args.type === 'mmi',
+      isTest: args.test,
+      shouldIncludeSnow: args.snow,
+    },
+  }),
+  new ManifestPlugin({
+    web_accessible_resources: webAccessibleResources,
+    manifest_version: MANIFEST_VERSION,
+    description: commitHash
+      ? `${args.env} build from git id: ${commitHash.substring(0, 8)}`
+      : null,
+    version: variables.get('METAMASK_VERSION') as string,
+    browsers: args.browser,
+    transform: args.lockdown
+      ? undefined
+      : (browserManifest) => {
+          // remove lockdown scripts from content_scripts
+          const mainScripts = browserManifest.content_scripts?.[0];
+          if (mainScripts) {
+            const keep = ['scripts/contentscript.js', 'scripts/inpage.js'];
+            mainScripts.js = mainScripts.js?.filter((js) => keep.includes(js));
+          }
+          return browserManifest;
+        },
+    zip: args.zip,
+    ...(args.zip
+      ? {
+          zipOptions: {
+            outFilePath: `../../../builds/metamask-[browser]-${variables.get(
+              'METAMASK_VERSION',
+            )}.zip`, // relative to output.path
+            mtime: getLatestCommit().timestamp(),
+            excludeExtensions: ['.map'],
+            // `level: 9` is the highest; it may increase build time by ~5% over level 1
+            level: 9,
+          },
+        }
+      : {}),
+  }),
+  // use ProvidePlugin to polyfill *global* node variables
+  new ProvidePlugin({
+    // Make a global `Buffer` variable that points to the `buffer` package.
+    Buffer: ['buffer', 'Buffer'],
+    // Make a global `process` variable that points to the `process` package.
+    process: 'process/browser',
+  }),
+  new CopyPlugin({
+    patterns: [
+      { from: join(context, '_locales'), to: '_locales' }, // translations
+      // misc images
+      // TODO: fix overlap between this folder and automatically bundled assets
+      { from: join(context, 'images'), to: 'images' },
+    ],
+  }),
+];
+// enable React Refresh in 'development' mode when `watch` is enabled
+if (__HMR_READY__ && isDevelopment && args.watch) {
+  const ReactRefreshWebpackPlugin: typeof ReactRefreshPluginType = require('@pmmmwh/react-refresh-webpack-plugin');
+  plugins.push(new ReactRefreshWebpackPlugin());
+}
+if (args.progress) {
+  const { ProgressPlugin } = require('webpack');
+  plugins.push(new ProgressPlugin());
+}
+// #endregion plugins
+
+const swcConfig = { args, safeVariables, browsersListQuery, isDevelopment };
+const tsxLoader = getSwcLoader('typescript', true, swcConfig);
+const jsxLoader = getSwcLoader('ecmascript', true, swcConfig);
+const ecmaLoader = getSwcLoader('ecmascript', false, swcConfig);
+
+const config = {
+  entry,
+  cache,
+  plugins,
+  context,
+  mode: args.env,
+  stats: args.stats ? 'normal' : 'none',
+  name: `MetaMask – ${args.env}`,
+  // use the `.browserlistrc` file directly to avoid browserslist searching
```

Comment:
We _can_ tell wepack to find a `browserslist` automatically... but when it does that it searches high and low for all the locations (even after it's found one it keeps searching). So it's faster to just tell it which file to use.